File: N8N/apps/workers/nodes/telegram.ts
import mustache from "mustache";
import fetch from "node-fetch";
import prisma from "../../../packages/db";

export default async function sendTelegramMessage(
  template: any,
  credentialId: string,
  Context: any
) {
  const credentials = await prisma.credentials.findMany({
    where: { id: credentialId },
  });
  if (!credentials) {
    throw new Error("Telegram credential was not found");
  }
  const data = credentials[0].data as { apiKey: string; chatId: string };
  const { apiKey, chatId } = data;
  if (!apiKey || !chatId) {
    throw new Error("Telegram credentials are needed");
  }
  const msg = mustache.render(template.message, Context);
  const response = await fetch(
    `https://api.telegram.org/bot${apiKey}/sendMessage`,
    {
      method: "POST",
      headers: { "Content-type": "application/json" },
      body: JSON.stringify({
        chat_id: chatId,
        text: msg,
      }),
    }
  );
  const message = await response.text();
  console.log("Message sent is: ", message);
  return { msg };
}

File: N8N/python-backend/Workers/nodes/telegram.py
from typing import Any, Dict

import httpx

node_details = {
    "type": "telegram",
    "name": "Telegram Bot",
    "description": "Send messages via Telegram Bot API",
    "category": "Communication",
    "icon": "📱"
}
import pystache
from db.database import get_db_session
from db.models.models import Credentials
from fastapi import HTTPException
from sqlmodel import select


async def send_Telegram_Msg(
    credential_id: str, template: Dict[str, Any], context: Dict[str, Any]
):
    db = get_db_session()
    statement = select(Credentials).where(Credentials.id == credential_id)
    credential = db.exec(statement).first()
    if not credential:
        raise HTTPException(status_code=400, detail="Telegram credential not found")
    data = credential.data
    api_Key: str = data.get("apiKey")
    chat_id: str = data.get("chatId")
    if not api_Key or not chat_id:
        raise HTTPException(
            status_code=404, detail="You have not provided both the botToken and chatId"
        )
    message_text = pystache.render(template.get("message", ""), context)
    url = f"https://api.telegram.org/bot{api_Key}/sendMessage"
    payload = {"chat_id": chat_id, "text": message_text}
    async with httpx.AsyncClient() as client:
        res = await client.post(url, json=payload)
        text = res.text
        if res.status_code != 200:
            raise HTTPException(status_code=502, detail="Telegram api error")
        return {"msg": message_text, "msg_sent": text}
